lib/babel.js

```diff
@@ -32,20 +34,46 @@ module.exports = {
 	}
 };
 
+function send(name, data) {
+	process.send({name: name, data: data});
+}
+
+process.on('uncaughtException', function (exception) {
+	send('uncaughtException', {uncaughtException: serializeValue(exception)});
+});
+
 var transpiled = babel.transformFileSync(testPath, options);
 requireFromString(transpiled.code, testPath, {
 	appendPaths: module.paths
 });
 
 if (!avaRequired) {
-	console.error('No tests found in ' + testPath + ', make sure to import "ava" at the top of your test file');
-	setImmediate(function () {
-		process.exit(1);
-	});
+	throw new Error('No tests found in ' + testPath + ', make sure to import "ava" at the top of your test file');
 }
 
 process.on('message', function (message) {
-	if (message['ava-kill-command']) {
+	var command = message['ava-child-process-command'];
+	if (command) {
+		process.emit('ava-' + command, message.data);
+	}
+});
+
+process.on('ava-kill', function () {
+	setTimeout(function () {
 		process.exit(0);
+	}, process.env.AVA_APPVEYOR ? 100 : 0);
```

Comment:
@sindresorhus got the intent right.
I originally had it as APPVEYOR, until a conversation with @BarryThePenguin 

The only tests this affects is ones that parse the stdout / stderr from our CLI.
There is a chance by running the tests fast, that they miss the last line or two (a summary or the failed stack trace). It's highly unlikely they will be grepping our code CLI output to get a pass/fail on their own tests.

They will still always get the correct exit code - they might just be missing the last "1 test failed".
If users report it as a problem it is easy enough to correct down the road.